org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameHandler.java

```diff
@@ -97,19 +95,21 @@ public void setEnabled(Object evaluationContext) {
 			IDocument document = LSPEclipseUtils.getDocument((ITextEditor) part);
 			if (document != null && isEnable) {
 				try {
-					isEnable = !LanguageServiceAccessor.getLanguageServers(document, LSPRenameHandler::isRenameProvider)
-							.get(50, TimeUnit.MILLISECONDS).isEmpty();
+					// See if there is a server that supports rename. We don't use LanguageServers.anyMatching in this
+					// case because we want to time out and not block the UI if we have to wait for servers to start
+					// but we also want to be able to detect the timeout and schedule a handler update for when
+					// the definitive info on server capabilities is available
+					isEnable = LanguageServers.forDocument(document).withFilter(LSPRenameHandler::isRenameProvider)
+							.computeFirst(ls -> CompletableFuture.completedFuture(true))
+							.get(50, TimeUnit.MILLISECONDS).isPresent();
 				} catch (java.util.concurrent.ExecutionException | TimeoutException e) {
-
-					// in case the language servers take longer to kick in, defer the enablement to
-					// a later time
-					LanguageServiceAccessor.getLanguageServers(document, LSPRenameHandler::isRenameProvider)
-							.thenAccept(languageServer -> {
-								boolean enabled = !languageServer.isEmpty();
-								final var handleEvent = new HandlerEvent(this, enabled, false);
-								fireHandlerChanged(handleEvent);
-							});
-
+					// Timed out, schedule a handler update from when all servers are started and have provided their capabilities
+					LanguageServers.forDocument(document).withFilter(LSPRenameHandler::isRenameProvider)
+						.computeFirst(ls -> CompletableFuture.completedFuture(true))
+						.thenAccept(o -> o.ifPresent(b -> {
+							final var handleEvent = new HandlerEvent(this, true, false);
+							fireHandlerChanged(handleEvent);
```

Comment:
Same here: Why do we rewrite this code? Does it have an advantage over the previous one? If not, and I would not know of any, I would leave the code as it was.